express/components/common.py

```diff
@@ -429,9 +486,15 @@ def run(cls) -> DataManifest:
             DataManifest: the output manifest
         """
         args = cls._parse_args()
+        if cls.data_sources_in is not None:
+            raise ValueError("Loader components should not have input data sources.")
         output_dataset_draft = cls.load(extra_args=json.loads(args.extra_args))
+        print("Output draft:", output_dataset_draft)
+        # verify outgoing data sources
+        cls._verify_data_sources_out(draft=output_dataset_draft)
```

Comment:
I think we still need a better view on how we want to convert the user dataframe to a new manifest, which is still an open topic in the dataset doc.